vissl/data/ssl_dataset.py

```diff
@@ -241,11 +265,17 @@ def __getitem__(self, idx):
             self._load_labels()
             self._labels_init = True
 
+        if self.data_limit >= 0:
+            self._init_subset()
```

Comment:
can we add the check `if not self._subset_initialized`: right here? so that it clear that we only call the init_subset if needed ? :)